silk/tests/test_events.py

```diff
@@ -81,19 +80,31 @@ def test_malformed_event(aggregator, instance, dd_run_check, file, log_warning,
 @pytest.mark.integration
 @pytest.mark.usefixtures('dd_environment')
 def test_events_test(aggregator, dd_run_check, instance):
-    # Freeze time when starting check to set initial time
-    with freeze_time("2012-01-13"):
+    # Mock timestamps instead of using freezegun
+    # First run to set the initial timestamp to 2012-01-13
+    with mock.patch('datadog_checks.silk.check.get_timestamp', return_value=1326412800) as mock_timestamp:
         check = SilkCheck('silk', {}, [instance])
+        assert check.latest_event_query == 1326412800
+        mock_timestamp.assert_called_once()
+        assert mock_timestamp() == 1326412800
 
-    # Freeze time running check initially to query for events from 1-13 to 1-14
-    with freeze_time("2012-01-14"):
+    # Second run to get events between 2012-01-13 and 2012-01-14 and set the timestamp to 2012-01-14
+    # Event 1 is between 2012-01-13 and 2012-01-14
+    with mock.patch('datadog_checks.silk.check.get_timestamp', return_value=1326499200) as mock_timestamp:
         dd_run_check(check)
         aggregator.assert_event("Event 1", count=1)
         aggregator.assert_event("Event 2", count=0)
-
-    # freeze time finally when running check 2nd time to query for events from 1-14 to 1-15
-    with freeze_time("2012-01-15"):
-        aggregator.reset()
+        assert check.latest_event_query == 1326499200
+        mock_timestamp.assert_called_once()
+        assert mock_timestamp() == 1326499200
+
+    # Third run to get events between 2012-01-14 and 2012-01-15
+    # Event 2 is between 2012-01-14 and 2012-01-15
+    aggregator.reset()
```

Comment:
I wonder if we can put this in a separate test or if we care about this specific sequence of check runs. Not a blocker

Reply:
I think we do care about it because essentially, the check works by recording the timestamps of each run for filtering. So we want to ensure that run 1 grabs events from 0 to 1. Then run 2 grabs 1 to 2 etc.